File: strategy/en_us/java/src/Converter.java
/**
 *	Interface which set the convert method to the classes which will
 *	implement their specific kind of a measure conversion.
 *	<p>
 *	The Context (@see MeasureCalculator) calls an Strategy (A class which
 *	implements this interface) to execute the action desirable (Convert a
 *	measure).
 */

public interface Converter {

	/**
	 *	Convert a measure of a type into another
	 *
	 *	@param 	currentMeasure	a measure
	 *	@return	the converted value of the measure
	 */
	public float convert(float currentMeasure);
}

File: strategy/en_us/python/converter.py
"""
There are a lot of ways to implement a Strategy Pattern in Python.
Since Interface does not exist in Python language, this example of
Strategy Pattern creates a Class with an "empty" method which will
be inherited by the classes which will implement that method.
"""

class Converter:
	"""
	The Context (measure_calculator) calls an Strategy (A class which
	inherit this class) to execute the action desirable (Convert a
	measure).
	"""

	def convert(current_measure):
		"""
		This method do nothing. The class which will inherit this class
		will be responsible to implement this method.
		"""
		pass
